p2p/dandelion-tower/src/pool/incoming_tx.rs

```diff
@@ -0,0 +1,110 @@
+//! Contains [`IncomingTx`] and [`IncomingTxBuilder`]
+use crate::{State, TxState};
+
+/// An incoming transaction that has gone through the preprocessing stage.
+pub struct IncomingTx<Tx, TxID, PID> {
+    /// The transaction.
+    pub(crate) tx: Tx,
+    /// The transaction ID.
+    pub(crate) tx_id: TxID,
+    /// The routing state of the transaction.
+    pub(crate) routing_state: TxState<PID>,
+}
+
+/// An [`IncomingTx`] builder.
+///
+// The const generics here are used to restrict what methods can be called.
+pub struct IncomingTxBuilder<const RS: bool, const DBS: bool, Tx, TxID, PID> {
```

Comment:
I created an incoming tx builder to force people to use the dandelion pool correctly.

I also tried a new technique to force people to use the builder correctly.